recipes/concurrencpp/all/conanfile.py

```diff
@@ -0,0 +1,122 @@
+    def _minimum_compilers_version(self):
+        return {"Visual Studio": "16", "msvc": "192", "clang": "11"}
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "20")
+
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler), False
+        )
+        if not minimum_version:
+            self.output.warn(
+                "concurrencpp requires C++20. Your compiler is unknown. Assuming it supports C++20."
+            )
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "concurrencpp requires clang >= 11 or Visual Studio >= 16.8.2 as a compiler!"
+            )
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx != "libc++":
+            raise ConanInvalidConfiguration("libc++ required")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake        
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):        
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"concurrencpp": "concurrencpp::concurrencpp"}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(self._module_subfolder,
+                            "conan-official-{}-targets.cmake".format(self.name))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "concurrencpp"
+        self.cpp_info.names["cmake_find_package_multi"] = "concurrencpp"
+        self.cpp_info.builddirs.append(self._module_subfolder)
+        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Comment:
I have to admit I copied the conan-official-{}-targets.cmake/find_package/find_package_multi parts from another recipe. Please let me know if it is correct as-is or should be changed.

The upstream library exports a `concurrencpp::concurrencpp` target: https://github.com/David-Haim/concurrencpp/blob/v.0.1.4/CMakeLists.txt#L120